mitmproxy-contentviews/src/protobuf/existing_proto_definitions.rs

```diff
@@ -151,11 +151,31 @@ impl std::fmt::Display for RpcInfo {
 fn parse_file_descriptor_set(definitions_path: &Path) -> anyhow::Result<Vec<FileDescriptor>> {
     let mut parser = Parser::new();
     parser.pure();
-    if let Some(parent) = definitions_path.parent() {
-        parser.include(parent);
+    if definitions_path.is_dir() {
+        walk_proto_directory(definitions_path, &mut parser)
+    } else {
+        if let Some(parent) = definitions_path.parent() {
+            parser.include(parent);
+        }
+        parser.input(definitions_path);
     }
-    parser.input(definitions_path);
     let fds = parser.file_descriptor_set()?;
     FileDescriptor::new_dynamic_fds(fds.file, &[])
         .context("failed to create dynamic file descriptors")
 }
+
+fn walk_proto_directory(definitions_path: &Path, parser: &mut Parser) {
+    parser.include(definitions_path);
+    for entry in definitions_path
+        .read_dir()
+        .expect("failed to read protobuf directory")
```

Comment:
...and change this to `.context("failed to read protobuf directory")?`. We don't want to crash the entire process if this fails.